src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -97,10 +97,13 @@
 
 (schema/defn handle-put-certificate-revocation-list!
   [incoming-crl-pem :- InputStream
-   {:keys [cacrl cacert ca-name]} :- ca/CaSettings]
+   {:keys [cacrl cacert enable-infra-crl infra-crl-path]} :- ca/CaSettings]
   (locking crl-write-serializer
     (try
-      (ca/update-crls incoming-crl-pem cacrl cacert)
+      (let [incoming-crls (utils/pem->crls incoming-crl-pem)]
```

Comment:
You can only read input streams once, so we read it here and pass the contents on to the `update-crls` calls.